### PR TITLE
OSHMEM: v1.3: adds shmem_fetch and shmem_set AMOs

### PR DESCRIPTION
**Pull request by alex-mikheev**
The commit adds atomic set and fetch functions as described in
oshmem 1.3 spec.

@igor-ivanov @jladd-mlnx please review

### CODE REVIEW

**Comment by jsquyres** [comment]
@alex-mikheev Please rebase this to the top of master to fix the Travis failure.  Please also set a milestone and appropriate labels.  Thanks!

File: oshmem/shmem/man/man3/Makefile.extra [hunk 1]
```diff
@@ -98,12 +98,26 @@ shmem_api_man_pages = \
...
         shmem/man/man3/shmem_long_fadd.3 \
         shmem/man/man3/shmem_longlong_fadd.3 \
+\
+		shmem/man/man3/shmem_double_fetch.3 \
+		shmem/man/man3/shmem_float_fetch.3 \
+		shmem/man/man3/shmem_int_fetch.3 \
+		shmem/man/man3/shmem_long_fetch.3 \
+		shmem/man/man3/shmem_longlong_fetch.3 \
+\
         shmem/man/man3/shmem_int_finc.3 \
```
Comment (igor-ivanov) [inline oshmem/shmem/man/man3/Makefile.extra]: @alex-mikheev fix alignment

File: oshmem/shmem/c/shmem_fetch.c [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        RUNTIME_CHECK_INIT();                                       \
+        RUNTIME_CHECK_PE(pe);                                       \
+        RUNTIME_CHECK_ADDR(target);                                 \
+                                                                    \
+        size = sizeof(out_value);                                   \
+        rc = MCA_ATOMIC_CALL(fadd(                                  \
+            (void*)target,                                          \
+            (void*)&out_value,                                      \
+            (const void*)&value,                                    \
+            size,                                                   \
```
Comment (alex-mikheev) [inline oshmem/shmem/c/shmem_fetch.c]: zero is added to the target

**Comment by igor-ivanov** [comment]
@alex-mikheev please ignore `I think that we should add shmemx_ functions` review general comment. Should be `I think that we should not add shmemx_ functions`

**Comment by alex-mikheev** [comment]
I added shmemx_ functions because we already have them for atomic swap. So it it reasonable to have such functions for the set and fetch. Or remove them all.

**Comment by jladd-mlnx** [comment]
:+1:  This looks good.

**Comment by alex-mikheev** [comment]
Actually we have shmemx_int(32|64) extensions for pretty much everything. If we want to remove them all it should be done in separate PR. Meanwhile it is reasonable that shmem_set/shmem_fetch also have such extenstion.

**Comment by igor-ivanov** [comment]
@alex-mikheev I do not know a reason we added shmemx_ functions in previous code. Do you know if there are request from someone? Personally I prefer to avoid having one w/o any needs.

**Comment by alex-mikheev** [comment]
i do not remember why it was added

**Comment by jladd-mlnx** [comment]
@igor-ivanov I'm going to merge this. If you guys decide to remove the `shmemx_` functions, do it as a separate PR.

**Pull request merged**